sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -454,145 +492,88 @@ export class StreamingReceiver extends MessageReceiver {
+        // the service does not send an error stating that the link is still open.
+        useNewName: true,
+        connectionId: this._context.connectionId,
+        onError: (args) => this._onError && this._onError(args)
+      });
 
-      // shall retry forever at an interval of 15 seconds if the error is a retryable error
-      // else bail out when the error is not retryable or the operation succeeds.
-      const config: RetryConfig<void> = {
-        operation: () =>
-          this.init(
-            // provide a new name to the link while re-connecting it. This ensures that
-            // the service does not send an error stating that the link is still open.
-            true
-          ).then(() => {
-            this._receiverHelper.addCredit(this.maxConcurrentCalls);
-            return;
-          }),
-        connectionId: connectionId,
-        operationType: RetryOperationType.receiverLink,
-        retryOptions: this._retryOptions,
-        connectionHost: this._context.config.host
-      };
-      // Attempt to reconnect. If a non-retryable error is encountered,
-      // retry will throw and the error will surface to the user's error handler.
-      await retry<void>(config);
+      this._receiverHelper.addCredit(this.maxConcurrentCalls);
+      logger.verbose(
+        `${this.logPrefix} onDetached: link has been reestablished, added ${this.maxConcurrentCalls} credits.`
+      );
     } catch (err) {
+      // the behavior of init() is such that it should _never_ throw an error. If it does then the library is doing
+      // something completely incorrect and we need to report it because it's unlikely that we'll properly
+      // recover from it.
       logger.logError(
         err,
-        "%s An error occurred while processing detached()",
-        this.logPrefix,
-        this.name,
-        this.address
+        `${this.logPrefix} onDetached: A fatal internal error has occurred. Connection will not be reestablished`
       );
-      if (typeof this._onError === "function") {
-        logger.verbose(`${this.logPrefix} Unable to automatically reconnect`);
-        try {
-          this._onError({
-            error: err,
-            errorSource: "receive",
-            entityPath: this.entityPath,
-            fullyQualifiedNamespace: this._context.config.host
-          });
-        } catch (err) {
-          logger.logError(
-            err,
-            `${this.logPrefix} User-code error in error handler called after disconnect`
-          );
-        } finally {
-          // Once the user's error handler has been called,
-          // close the receiver to prevent future messages/errors from being received.
-          // Swallow errors from the close rather than forwarding to user's error handler
-          // to prevent a never ending loop.
-          await this.close().catch(() => {});
-        }
-      }
+
+      this._onError &&
+        this._onError({
+          errorSource: "internal",
```

Comment:
I realized that we will hit this code path when the user makes use of the abort signal. In which case the error source is not really internal...

Instead of throwing the AbortError from inside `init()`, should the error be simply reported to the user's error handler and then we do nothing else?

Reply:
On thinking about this more I realized that the error handling here really is just completely mirroring what I was doing with init() and that we have sufficient coverage of both init() and onDetached() so I don't need it in both spots.

So I:
1. Removed the catch entirely from onDetached around init()
2. Removed the internal eventSource since we didn't need it anymore (this was the only spot it was used)
3. Adjusted the catch so we definitely onError(abortError) before throwing so the user knows the loop is dead (and why).
4. Changed our tests to check for it.

I think we're basically in the same spot. I'm still okay with throwing from init() since aborterror is one of the few user-facing errors that we do throw purposefully.